README.rst

```diff
@@ -16,6 +16,19 @@ Requirements
 
 Requires Python 3.6 or above.
```

Comment:
@z3z1ma were you. using 3.6 locally for developing this? Just double-checking, as I was not and there's some chance that some of the syntax we use in the type hints that we introduce require higher versions.

Reply:
I was not using 3.6 but I am going to fallback and adjust the type hinting as per your suggestions to maintain wider compatibility for users @fernandobrito .